src/testcafe-runner.js

```diff
@@ -223,23 +58,184 @@ async function runReporter ({ suiteName, results, metrics, assetsPath, browserNa
+  }
+  if (suite.stopOnFirstFail) {
+    cli.push('--stop-on-first-fail');
+  }
+  if (suite.disablePageCaching) {
+    cli.push('--disable-page-caching');
+  }
+  if (suite.disableScreenshots) {
+    cli.push('--disable-screenshots');
+  }
+  if (suite.quarantineMode) {
+    const flags = [];
+    if (suite.quarantineMode.attemptLimit) {
+      flags.push(`attemptLimit=${suite.quarantineMode.attemptLimit}`);
+    }
+    if (suite.quarantineMode.successThreshold) {
+      flags.push(`successThreshold=${suite.quarantineMode.successThreshold}`);
+    }
+    if (flags.length) {
+      cli.push('--quarantine-mode', flags.join(','));
+    }
+  }
+
+  // Record a video if it's not a VM or if SAUCE_VIDEO_RECORD is set
+  const shouldRecordVideo = !suite.disableVideo && (!process.env.SAUCE_VM || process.env.SAUCE_VIDEO_RECORD);
+  if (shouldRecordVideo) {
+    cli.push(
+      '--video', assetsPath,
+      '--video-options', 'singleFile=true,failedOnly=false,pathPattern=video.mp4',
+    );
+  }
+
+  // Screenshots
+  if (suite.screenshots) {
+    // Set screenshot pattern as fixture name, test name and screenshot #
+    // This format prevents nested screenshots and shows only the info that
+    // a Sauce session needs
+    const pathPattern = '${FIXTURE}__${TEST}__screenshot-${FILE_INDEX}';
+    const takeOnFails = suite.screenshots.takeOnFails;
+    const fullPage = suite.screenshots.fullPage;
+    cli.push('--screenshots', `takeOnFails=${takeOnFails},fullPage=${fullPage},path=${assetsPath},pathPattern=${pathPattern}`);
+  }
+
+  if (process.env.HTTP_PROXY) {
+    const proxyURL = new URL(process.env.HTTP_PROXY);
+    cli.push('--proxy', proxyURL.host);
+  }
+
+  // Filters
+  if (suite.filter && suite.filter.test) {
+    cli.push('--test', suite.filter.test);
+  }
+  if (suite.filter && suite.filter.fixture) {
+    cli.push('--fixture', suite.filter.fixture);
+  }
+  if (suite.filter && suite.filter.testGrep) {
+    cli.push('--test-grep', suite.filter.testGrep);
+  }
+  if (suite.filter && suite.filter.fixtureGrep) {
+    cli.push('--fixture-grep', suite.filter.fixtureGrep);
+  }
+  if (suite.filter && suite.filter.testMeta) {
+    const filters = [];
+    for (const key of Object.keys(suite.filter.testMeta)) {
+      filters.push(`${key}=${suite.filter.testMeta[key]}`);
+    }
+    cli.push('--test-meta', filters.join(','));
+  }
+  if (suite.filter && suite.filter.fixtureMeta) {
+    const filters = [];
+    for (const key of Object.keys(suite.filter.fixtureMeta)) {
+      filters.push(`${key}=${suite.filter.fixtureMeta[key]}`);
+    }
+    cli.push('--fixture-meta', filters.join(','));
+  }
+
+  // Reporters
+  const xmlReportPath = path.join(assetsPath, 'report.xml');
+  const jsonReportPath = path.join(assetsPath, 'report.json');
+  cli.push('--reporter', `xunit:${xmlReportPath},json:${jsonReportPath},list`);
+
+  return cli;
 }
 
-async function run (runCfgPath, suiteName) {
+async function runTestCafeV2 (runCfgPath, suiteName) {
```

Comment:
Is that the final name? 😅

Reply:
It has been renamed, and the actual execution part has been moved to a dedicated function 😄